src/odemis/acq/fastem.py

```diff
@@ -962,49 +963,60 @@ def estimateTiledAcquisitionTime(stream, stage, area):
-    # correctly and without gap.
-    overlap = STAGE_PRECISION / stream.emitter.horizontalFoV.value
-    logging.debug("Overlap is %s%%", overlap * 100)  # normally < 1%
+        # The stage movement precision is quite good (just a few pixels). The stage's
+        # position reading is much better, and we can assume it's below a pixel.
+        # So as long as we are sure there is some overlap, the tiles will be positioned
+        # correctly and without gap.
+        overlap = STAGE_PRECISION / stream.emitter.horizontalFoV.value
+        logging.debug("Overlap is %s%%", overlap * 100)  # normally < 1%
 
-    def _pass_future_progress(sub_f, start, end):
-        f.set_progress(start, end)
+        def _pass_future_progress(sub_f, start, end):
+            f.set_progress(start, end)
 
-    # Note, for debugging, it's possible to keep the intermediary tiles with log_path="./tile.ome.tiff"
-    sf = stitching.acquireTiledArea([stream], stage, area, overlap, registrar=REGISTER_IDENTITY,
-                                    focusing_method=FocusingMethod.NONE)
-    # Connect the progress of the underlying future to the main future
-    # FIXME removed to provide better progress update in GUI
-    #  When _tiledacq.py has proper time update implemented, add this line here again.
-    # sf.add_update_callback(_pass_future_progress)
-    das = sf.result()
+        # Note, for debugging, it's possible to keep the intermediary tiles with log_path="./tile.ome.tiff"
+        self.sf = stitching.acquireTiledArea([stream], stage, area, overlap, registrar=REGISTER_IDENTITY,
+                                        focusing_method=FocusingMethod.NONE)
+        # Connect the progress of the underlying future to the main future
+        # FIXME removed to provide better progress update in GUI
+        #  When _tiledacq.py has proper time update implemented, add this line here again.
+        # sf.add_update_callback(_pass_future_progress)
+        # Go one level deeper to cancel the future of the underlying acquisition
+        # TiledAcquisitionTask._cancelAcquisition will hence be called
+        f.task_canceller = self._cancel_acquisition
+        das = self.sf.result()
 
-    if len(das) != 1:
-        logging.warning("Expected 1 DataArray, but got %d: %r", len(das), das)
+        if len(das) != 1:
+            logging.warning("Expected 1 DataArray, but got %d: %r", len(das), das)
 
-    # Switch immersion mode back on, so we can focus the SEM from the TFS GUI.
-    stream.emitter.immersion.value = True
+        # Switch immersion mode back on, so we can focus the SEM from the TFS GUI.
+        stream.emitter.immersion.value = True
 
-    # FIXME auto blanking not working properly, so force beam blanking after image acquisition for now.
-    stream.emitter.blanker.value = True
+        # FIXME auto blanking not working properly, so force beam blanking after image acquisition for now.
+        stream.emitter.blanker.value = True
```

Comment:
These lines should actually be in a (try)/finally block. So even if the sub future fails or is cancelled the hardware is still reset to a proper state. (That should have been done in the original code, but now it's even more important, as cancelling is going to be much more common that acquisition failure, hopefully!)